eng/pipelines/templates/steps/code-coverage.yml

```diff
@@ -0,0 +1,22 @@
+steps:
+  - task: Maven@3
+    displayName: 'Generate aggregate code coverage report'
+    condition: eq(variables['Coverage'], 'true')
+    inputs:
+      mavenPomFile: sdk/${{ parameters.ServiceDirectory }}/pom.xml
+      options: -Pcoverage
+      mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
+      javaHomeOption: 'JDKVersion'
+      jdkVersionOption: $(JavaTestVersion)
+      jdkArchitectureOption: 'x64'
+      goals: jacoco:report-aggregate
+
+  # Azure DevOps only seems to respect the last code coverage result published, so only do this for Windows + Java LTS.
+  # Code coverage reporting is setup only for Track 2 modules.
+  - task: PublishCodeCoverageResults@1
+    condition: eq(variables['Coverage'], 'true')
+    inputs:
+      codeCoverageTool: JaCoCo
+      summaryFileLocation: sdk/${{ parameters.ServiceDirectory }}/target/site/jacoco-aggregate/jacoco.xml
```

Comment:
This will work for jacoco:report-aggregate but may not work for communication since each sdk in communication has its own pipeline and it runs jacoco:report instead of jacoco:report-aggregate.  The summaryFileLocation will be inside the sdk instead of the serviceDirectory